test/ng/compileSpec.js

```diff
@@ -7815,4 +7817,125 @@ describe('$compile', function() {
         expect(element.hasClass('fire')).toBe(true);
       }));
   });
+
+  describe('element replacement', function() {
+    it('should broadcast $destroy only on removed elements, not replaced', function() {
```

Comment:
I'm tempted to just drop this test for now. The cases when `$destroy` should/should-not fire are too unclear and I think the current implementation is wrong (or at least super weird) so I'm not sure if I want to enforce that with a test. I think all the other tests are good though (ensuring all elements get passed to `cleanData`).